src/main/java/org/jabref/gui/commonfxcontrols/FieldFormatterCleanupsPanel.java

```diff
@@ -0,0 +1,139 @@
+package org.jabref.gui.commonfxcontrols;
+
+import javax.inject.Inject;
+
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.ListProperty;
+import javafx.beans.property.ReadOnlyObjectWrapper;
+import javafx.fxml.FXML;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.SelectionMode;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableView;
+import javafx.scene.input.KeyCode;
+import javafx.scene.layout.VBox;
+
+import org.jabref.gui.StateManager;
+import org.jabref.gui.icon.IconTheme;
+import org.jabref.gui.util.FieldsUtil;
+import org.jabref.gui.util.ValueTableCellFactory;
+import org.jabref.gui.util.ViewModelListCellFactory;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.cleanup.FieldFormatterCleanup;
+import org.jabref.model.cleanup.Formatter;
+import org.jabref.model.entry.field.Field;
+
+import com.airhacks.afterburner.views.ViewLoader;
+
+public class FieldFormatterCleanupsPanel extends VBox {
+
+    @FXML private CheckBox cleanupsEnabled;
+    @FXML private TableView<FieldFormatterCleanup> cleanupsList;
+    @FXML private TableColumn<FieldFormatterCleanup, Field> fieldColumn;
+    @FXML private TableColumn<FieldFormatterCleanup, Formatter> formatterColumn;
+    @FXML private TableColumn<FieldFormatterCleanup, Field> actionsColumn;
+    @FXML private ComboBox<Field> addableFields;
+    @FXML private ComboBox<Formatter> addableFormatters;
+
+    @Inject private StateManager stateManager;
```

Comment:
I think it should work if you pass the stateManager as ctor parameter instead of using the injection

Reply:
The FieldFormatterCleanupsPanel is instantiated by the fxml file. I could theoretically pass an argument here using the `@NamedArg`-annotation. But passing a StateManager seems to me not possible that way. So the alternative would be to hardcode this in the view.